backend/geonature/core/auth/routes.py

```diff
@@ -112,6 +112,15 @@ def loginCas():
                 "id_organisme": organism_id,
             }
             response.set_cookie("current_user", str(current_user), expires=cookie_exp)
+
+            # Log the user in
+            user = db.session.execute(
+                sa.select(models.User)
+                .where(models.User.identifiant == current_user["user_login"])
+                .where(models.User.filter_by_app())
+            ).scalar_one()
+            login_user(user)
```

Comment:
**Suggestion:** It's a good practice to handle exceptions when interacting with the database. Consider adding a try-except block around the `db.session.execute()` call to handle any potential SQLAlchemy or database errors. [best practice]
```suggestion
            try:
                user = db.session.execute(
                    sa.select(models.User)
                    .where(models.User.identifiant == current_user["user_login"])
                    .where(models.User.filter_by_app())
                ).scalar_one()
                login_user(user)
            except Exception as e:
                log.error(f"Error logging in user: {e}")
                raise
```